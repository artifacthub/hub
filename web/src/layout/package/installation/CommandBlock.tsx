import React from 'react';
import SyntaxHighlighter from 'react-syntax-highlighter';
import { docco } from 'react-syntax-highlighter/dist/cjs/styles/hljs';

import ButtonCopyToClipboard from '../../common/ButtonCopyToClipboard';
import styles from './ContentInstall.module.css';

interface Props {
  command: string;
  language?: string;
  title?: string;
  filename?: string;
}

const CommandBlock = (props: Props) => (
  <>
    {props.title && (
      <div className="my-2">
        <small className="text-muted mt-2 mb-1">{props.title}</small>
      </div>
    )}

    {props.filename && (
      <div className="mb-2">
        <span className="badge badge-dark badge-sm">
          <small className="text-uppercase mr-2">File:</small>
          {props.filename}
        </span>
      </div>
    )}

    <div className="d-flex align-items-start">
      <div className={`flex-grow-1 mr-3 ${styles.blockWrapper}`}>
        <SyntaxHighlighter
          language={props.language || 'bash'}
          style={docco}
          customStyle={{
            backgroundColor: 'var(--color-1-10)',
          }}
        >
          {props.command}
        </SyntaxHighlighter>
      </div>

      <div>
        <ButtonCopyToClipboard text={props.command} className={`btn-primary rounded-circle ${styles.copyBtn}`} />
      </div>
    </div>
  </>
);

export default CommandBlock;
