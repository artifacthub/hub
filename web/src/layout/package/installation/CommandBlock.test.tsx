import { render } from '@testing-library/react';
import React from 'react';

import CommandBlock from './CommandBlock';

describe('CommandBlock', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('creates snapshot', () => {
    const result = render(<CommandBlock command="command" />);
    expect(result.asFragment()).toMatchSnapshot();
  });

  describe('Render', () => {
    it('renders component', () => {
      const { getByText } = render(<CommandBlock command="command" />);

      expect(getByText('command')).toBeInTheDocument();
    });

    it('renders component with title', () => {
      const { getByText } = render(<CommandBlock title="this is a title" command="command" />);

      expect(getByText('this is a title')).toBeInTheDocument();
    });

    it('renders component with filename', () => {
      const { getByText } = render(<CommandBlock filename="filename" command="command" />);

      expect(getByText('File:')).toBeInTheDocument();
      expect(getByText('filename')).toBeInTheDocument();
    });
  });
});
