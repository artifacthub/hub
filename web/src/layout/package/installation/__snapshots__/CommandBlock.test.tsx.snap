// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandBlock creates snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-flex align-items-start"
  >
    <div
      class="flex-grow-1 mr-3 blockWrapper"
    >
      <pre
        style="display: block; overflow-x: auto; padding: 0.5em; color: rgb(0, 0, 0); background: rgb(248, 248, 255);"
      >
        <code
          class="language-bash"
        >
          <span
            style="color: rgb(0, 134, 179);"
          >
            command
          </span>
        </code>
      </pre>
    </div>
    <div>
      <div
        class="position-relative undefined"
      >
        <button
          class="btn btn-sm btn-primary rounded-circle copyBtn"
          data-testid="ctcBtn"
          type="button"
        >
          <div
            class="d-flex flex-row align-items-center"
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="13"
                rx="2"
                ry="2"
                width="13"
                x="9"
                y="9"
              />
              <path
                d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
              />
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
